pkg/image/prune/imagepruner.go

```diff
@@ -491,7 +491,23 @@ func (p *imagePruner) Run(pruneImage ImagePruneFunc, pruneStream ImageStreamPrun
 	pruneImages(p.g, imageNodes, pruneImage, pruneStream, pruneManifest)
 
 	layerNodes := layerNodeSubgraph(allNodes)
-	pruneLayers(p.g, layerNodes, pruneLayer, pruneBlob)
+	registryURL := ""
+	if len(imageNodes) > 0 {
+		// we only support a single internal registry, and all images should have the same registry
+		// so we just take the 1st one and use it
+		if ref, err := imageapi.ParseDockerImageReference(imageNodes[0].Image.DockerImageReference); err == nil {
+			registryURL = ref.Registry
+		} else {
+			glog.Errorf("error determining registry URL from %q: %v", imageNodes[0].Image.DockerImageReference, err)
+		}
+	}
+
+	if len(registryURL) == 0 {
```

Comment:
I liked the impl with this check in the `pruneLayers` method.  I just wasn't sure I wanted a utility method handling my logging for me.  I guess I could be happier with this in `pruneLayers` and it using `util.HandleError`.

It just felt weird for me to have a method that failed at its task not indicate that failure to the caller.